cmd/ib-sriov-cni/main.go

```diff
@@ -164,15 +160,82 @@ func cmdAdd(args *skel.CmdArgs) error {
 		}()
 	}
 
-	result := &current.Result{}
-	result.Interfaces = []*current.Interface{{
-		Name:    args.IfName,
-		Sandbox: netns.Path(),
-	}}
-
 	err = sm.SetupVF(netConf, args.IfName, args.ContainerID, netns)
+	if err != nil {
+		return fmt.Errorf("failed to set up pod interface %q from the device %q: %v", args.IfName, netConf.Master, err)
+	}
+	return nil
+}
+
+// Run the IPAM plugin
+func runIPAMplugin(args *skel.CmdArgs, netConf *localtypes.NetConf, netns ns.NetNS, result **current.Result) error {
+	if netConf.IPAM.Type == ipamDHCP {
+		return fmt.Errorf("ipam type dhcp is not supported")
+	}
+	r, err := ipam.ExecAdd(netConf.IPAM.Type, args.StdinData)
+	if err != nil {
+		return fmt.Errorf("failed to set up IPAM plugin type %q from the device %q: %v",
+			netConf.IPAM.Type, netConf.Master, err)
+	}
+
+	defer func() {
+		if err != nil {
+			_ = ipam.ExecDel(netConf.IPAM.Type, args.StdinData)
+		}
+	}()
+
+	// Convert the IPAM result into the current Result type
+	var newResult *current.Result
+	newResult, err = current.NewResultFromResult(r)
+	if err != nil {
+		return err
+	}
+
+	if len(newResult.IPs) == 0 {
+		return errors.New("IPAM plugin returned missing IP config")
+	}
+
+	newResult.Interfaces = (*result).Interfaces
+
+	for _, ipc := range newResult.IPs {
+		// All addresses apply to the container interface (move from host)
+		ipc.Interface = current.Int(0)
+	}
+
+	err = netns.Do(func(_ ns.NetNS) error {
```

Comment:
it better to use this method https://github.com/containernetworking/plugins/blob/47a9fd80c8258a288db0bb39f68516a4770012a4/pkg/ns/ns_linux.go#L222

Reply:
they achieve the same thing.

with NetNS interface you have a network namespace file descriptor inside which is kept open, this allows you to `Do` several operations before closing the file.

so i think you can either create it once and have it used throughout the CmdAdd flow (create once at top level and defer its close) or perform each individual operation separately using `WithNetNSPath`.

downside of the latter approach is performing multiple open/close file operations

Reply:
you right let keep it as is